File: Wirtualny świat/C++/ConsoleApplication18/guarana.cpp
#include "stdafx.h"
#include "guarana.h"


guarana::guarana(swiat& ref, int x, int y) :roslina(ref, x, y)
{
	this->sila = 0;
	this->inicjatywa = 0;
	this->nazwa = 'G';
	this->specjalna_kolizja = 1;
}

char guarana::rysowanie() {
	return this->nazwa;
}
void guarana::nowe(swiat& ref, int x, int y)
{
	organizm *o = new guarana(ref, x, y);
	cout << "dodano: " << this->rysowanie() << " na polu: " << x + 1 << ':' << y + 1 << ":\n";
}
void guarana::spec_kolizja(organizm* org) {
	org->ustaw_sila_organizmu(org->sila_organizmu() + 3);
	cout << org->rysowanie() << " zwiekszyl swa sile o 3\n";
}
guarana::~guarana()
{
}

File: Wirtualny świat/java/src/organizm/guarana.java

package organizm;
import java.util.Random;
import swiat.swiat;


public class guarana extends roslina {
    public guarana(swiat WielkiSwiat, int x, int y)
    {
        super(WielkiSwiat, x, y);
    	this.nazwa = 'G';
        this.specjalna_kolizja = true;
    }
    @Override
  public void spec_kolizja(organizm org) {
	org.ustaw_sila_organizmu(org.sila_organizmu() + 3);
        System.out.println(org.rysowanie() + " na polu: " + (x+1) + ":" + (y+1) + " zwiekszyl swoja sile o 3");
}
    public void nowy(swiat WielkiSwiat, int x, int y)
    {
        organizm o = new guarana(WielkiSwiat, x, y);
    }


}